ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantFeature.scala

```diff
@@ -26,6 +27,7 @@ object Features {
       selfServiceErrorCodes = experimental.flatMap(_.selfServiceErrorCodes).isDefined,
       userManagement = features.flatMap(_.userManagement).isDefined,
       commandDeduplicationFeatures = response.getFeatures.getCommandDeduplication,
+      staticTime = experimental.flatMap(_.staticTime).isDefined,
```

Comment:
I think I'm confused a bit about the usage of `isDefined` here. Shouldn't it be `exists(_ == true)`? Wouldn't this return true even if you set static time to false in the message?

I understand that it works because of the way we wire these settings in the runners, but it's quite confusing.

Reply:
@nicu-da : AFAIR, `experimental.flatMap(_.staticTime)` is of type `Option[ExperimentalStaticTime]`, and the feature is on if that value is not `None`. 

In that case isn't `isDefined` the more direct way to detect that than `exists(_ == true)`, which BTW does not typecheck for me. Put differently, can you explain what confuses you?

Reply:
@meiersi-da Indeed, `exists(_ == true)` seems to not be possible because we don't actually check the field value but the existence of a given message of type. 

In my own opinion, this is not just confusing but also contra-intuitive. We should be explicit in our messages and not infer if the features are enabled/disabled from the presence of a protobuf message.
Even more, as a developer and user of the API I would expect the exact same behavior if the message is present or not.

Therefore I would expect that if I send a response with the following value:
`ExperimentalFeatures()`
it would behave exactly as 
`ExperimentalFeatures(staticTime = Some(StaticTime()))`

I would recommend that we add an `enabled` field to the static time message (and maybe the other existing messages as well for consistency), so that we are explicit in our intentions.

Reply:
I perceive the current implementation of features list like a glorified list of enums. One could also implement it as you suggest more in the vain of a bitmask (with fields on/off). Both of these approaches have their merits and can confuse the other part of the population that expects the other approach to prevail.

Reply:
@mziolekda I would not expect anyone to be confused by the meaning of `StaticTime(enabled = false)`, and what it would represent. I would however see reasons to be confused by an empty message which is used as a boolean. 

Now I'm not going to touch the idea that in other languages this might be different, as with protobuf it could use the default value for the field, something that is possible also with scalapb by using `no_box` in which case there would be no way of figuring out what do we want from this field. 
For me, the confusion is even more evident if we think as to why we have the messages. 

But the simple fact that if we pass `StaticTime.defaultInstance` as a response it in turn `enables` the static time functionality for me it works in the opposite way of how it should sanely work.

This would in turn become even more confusing if we start adding fields to the `StaticTime` message, which I assume it's the whole reason why we don't just use a boolean.

Reply:
@mziolekda Had a chat with @meiersi-da, it's all good with this PR,  I will tackle some small changes to improve on these descriptors in another PR.